crates/common/src/compile.rs

```diff
@@ -300,12 +331,16 @@ const CONTRACT_RUNTIME_SIZE_LIMIT: usize = 24576;
 // https://eips.ethereum.org/EIPS/eip-3860
 const CONTRACT_INITCODE_SIZE_LIMIT: usize = 49152;
 
+const REVIVE_INITCODE_SIZE_LIMIT: usize = 256000;
+
 /// Contracts with info about their size
 pub struct SizeReport {
     /// What kind of report to generate.
     report_kind: ReportKind,
     /// `contract name -> info`
     pub contracts: BTreeMap<String, ContractInfo>,
+    /// Whether contracts where compiled with revive
+    pub revive_compile: bool,
```

Comment:
I do not think this is a right approach to extend the structs with `revive_compile` to add revive dependent functionality

Reply:
I saw it better to keep all logic for checking if we have surpassed the contract size inside the SizeReport for revive since thats where the other code is i also saw matter labs took a similar approach do you see a better way?

Reply:
@smiasojed ive changed the approach please review it